connect/tls.go

```diff
@@ -56,7 +56,7 @@ type verifierFunc func(cfg *tls.Config, rawCerts [][]byte) error
 
 // defaultTLSConfig returns the standard config with no peer verifier. It is
 // insecure to use it as-is.
-func defaultTLSConfig() *tls.Config {
+func defaultTLSConfig(alpnNextProtos *[]string) *tls.Config {
```

Comment:
Is it still the default if it's modified by user options 😄 ?

I think it would be a bit cleaner if we left this method alone and instead just set the non-default option in `NewServiceWithConfig` if it's set.